src/components/molecules/forms/jobForm/JobForm.tsx

```diff
@@ -83,102 +90,142 @@ const JobForm = () => {
     }
   };
 
+  const preview = () => {
```

Comment:
Is it possible to combine this with the almost exact same object in `submit`? If I see correctly, the only difference is the check for empty string vs not in the `submit` function. Is this something we should also check for in `submit`? I think it would be cleaner to return this from a function that takes in field as a parameter and reuse it in submit and here if possible

Reply:
I dont know it would be better as the submit performs validations and to combine the functions the preview needs to be done before validations making only the data object reusable. Furthermore, the preview has to specifically set optional values to undefined while the request does not as the API handles set invalid dates to undefined

Reply:
> I dont know it would be better as the submit performs validations and to combine the functions the preview needs to be done before validations making only the data object reusable.

I am not sure what you mean?

> Furthermore, the preview has to specifically set optional values to undefined while the request does not as the API handles set invalid dates to undefined

But does the API complain if it get's `undefined` as a value?

My thought was:


```javascript
const getJob(field) => {
  return {
    company: fields['company']?.value,
    title: fields['title']?.value,
    type: fields['type']?.value,
    tags: fields['tags']?.value.split(' '),
    description_preview: fields['description_preview'].value,
    description: fields['description'].value,
    start_date:
    fields['start_date'].value !== ''
      ? new Date(fields['start_date'].value)
      : undefined,
    location: fields['location'].value,
    link: fields['link'].value,
    due_date:
    fields['due_date'].value !== ''
      ? new Date(fields['due_date'].value)
      : undefined,

    published_date: new Date(),
  }
}

// In submit
const resp = await createJob(getJob(fields));

// In preview
const data = getJob() as JobItem;
```

Reply:
i see i thought you ment combining submit and preview into one submit function. Jepp i can add that into the JobForm